spec/formatting.md

```diff
@@ -8,6 +8,15 @@ when formatting a message for display in a user interface, or for some later pro
 The document is part of the MessageFormat 2.0 specification,
 the successor to ICU MessageFormat, henceforth called ICU MessageFormat 1.0.
 
+## Literal Resolution
+
+The resolved value of _text_, _literal_ and _nmtoken_ tokens
+is always a string concatenation of its parts,
+with escape sequences resolving to their escaped characters.
+When a literal value is used as a formatting function argument or option value,
+the formatting function MUST treat option values the same independently of their presentation,
+such that e.g. the options `foo=42` and `foo=|42|` have the same effect.
```

Comment:
I think this might not be correct or that it has the potential to be incorrect?

The reason we define `nmtoken` separate from `literal` (or `unquoted`) is because it fulfills a separate role. For example, in a `plural` selector, the `key` values can include `nmtoken` values from an array of keywords (`zero`, `one`, `two`, `few`, `many`, `other`/`*`) and *separately* certain literal values. A formatter such as the number formatter might accept a number of `nmtoken` keyword arguments (`integer`, `percent`) but might also allow literals in the same argument. 

Admittedly I can't think of a use case like this currently and it would be a poor formatter design that depended solely on the difference between `integer` and `|integer|` to know if the value were meant to be a token or a string. In fact, the more I look at this, the more I tend to think that `nmtoken` might need to go (reserving literal values as keywords in `key` and `option` could be done in the registry)? It would certainly simplify the ABNF. 

The downside of removing `nmtoken` is that the `nmtoken` production allows `key` values that start with `-` and `:`, such as `when -42` or `when :foo`, without quoting the values (`|-42|` and `|:foo|`). I don't care about `:` (in fact, I think it's confusing to allow it), but the minus sign feels important for operating with numbers.

Reply:
> In fact, the more I look at this, the more I tend to think that `nmtoken` might need to go (reserving literal values as keywords in `key` and `option` could be done in the registry)? It would certainly simplify the ABNF.

Could you clarify whether you're suggesting that the registry ought to influence what's valid syntax? That would be rather problematic.

Reply:
Agreed. That's included in the current registry PR #368 as e.g. `<match values="one other"/>`.
https://github.com/unicode-org/message-format-wg/blob/39f13a99db4f0391ebabd8534dbf54e190c6c827/spec/registry.dtd#L34-L36